core/lib/multivm/src/tracers/call_tracer/era_vm/mod.rs

```diff
@@ -0,0 +1,192 @@
+use era_vm::{
+    opcode::{RetOpcode, Variant},
+    value::FatPointer,
+    Execution, Opcode,
+};
+use zksync_state::ReadStorage;
+use zksync_types::{
+    vm_trace::{Call, CallType},
+    zk_evm_types::FarCallOpcode,
+    CONTRACT_DEPLOYER_ADDRESS, U256,
+};
+
+use super::CallTracer;
+use crate::{
+    era_vm::tracers::traits::{Tracer, VmTracer},
+    interface::VmRevertReason,
+};
+
+impl Tracer for CallTracer {
+    fn after_execution(
+        &mut self,
+        opcode: &Opcode,
+        execution: &mut era_vm::Execution,
+        _state: &mut era_vm::state::VMState,
+    ) {
+        match opcode.variant {
+            Variant::NearCall(_) => {
+                self.increase_near_call_count();
+            }
+            Variant::FarCall(far_call) => {
+                // We use parent gas for properly calculating gas used in the trace.
+                let current_ergs = execution.gas_left().unwrap();
+                let parent_gas = execution
+                    .running_contexts
+                    .last()
+                    .map(|call| call.frame.gas_left.0.saturating_add(current_ergs))
+                    .unwrap_or(current_ergs) as u64;
+
+                // we need to to this cast because `Call` uses another library
+                let far_call_variant = match far_call as u8 {
+                    0 => FarCallOpcode::Normal,
+                    1 => FarCallOpcode::Delegate,
+                    2 => FarCallOpcode::Mimic,
+                    _ => unreachable!(),
```

Comment:
We can avoid the unreachable! macro here, let's use the variant defined here on zksync-era to match with ours, even if we have to use total namespaces.